harness/determined/exec/launch.py

```diff
@@ -0,0 +1,98 @@
+import copy
+import json
+import logging
+import re
+import subprocess
+import sys
+from typing import Dict
+
+import determined as det
+import determined.common
+from determined.common import constants, storage
+
+
+def match_legacy_trial_class(arg: str) -> bool:
+    trial_class_regex = re.compile("^[a-zA-Z0-9_.]+:[a-zA-Z0-9_]+$")
+    if trial_class_regex.match(arg):
+        return True
+    return False
+
+
+def launch(experiment_config: det.ExperimentConfig) -> int:
+    # If native is enabled, harness will load from native entrypoint command
+    if experiment_config.native_enabled():
+        from determined.exec import harness
+
+        return harness.main(train_entrypoint=None)
+
+    entrypoint = experiment_config.get_entrypoint()
+    slots_per_trial = experiment_config.slots_per_trial()
+
+    if not entrypoint:
+        logging.error("Entrypoint not specified in experiment config")
+        return 1
+
+    if isinstance(entrypoint, list):
+        entrypoint = " ".join(entrypoint)
+
+    if match_legacy_trial_class(entrypoint):
+        # Legacy entrypoint ("model_def:Trial" or ["model_def:Trial"]) detected
```

Comment:
wait, why would we let somebody use the new format (`["model_def:Trial"]`) with the legacy behavior?

Reply:
well, i guess no one would/should do this. it seemed like a natural case to support since entrypoint can be a list or string type now, but it doesn't make sense to explicitly allow it for the legacy use case. i've removed it.